pkg/forwarder/envelope.go

```diff
@@ -0,0 +1,46 @@
+package forwarder
+
+import (
+	"encoding/json"
+
+	"github.com/ThreeDotsLabs/watermill"
+	"github.com/ThreeDotsLabs/watermill/message"
+	"github.com/pkg/errors"
+)
+
+// messageEnvelope wraps Watermill message and contains destination topic.
+type messageEnvelope struct {
+	DestinationTopic string `json:"destination_topic"`
+
+	UUID     string            `json:"uuid"`
+	Payload  []byte            `json:"payload"`
+	Metadata map[string]string `json:"metadata"`
+}
+
+func wrapMessageInEnvelope(destinationTopic string, msg *message.Message) (*message.Message, error) {
+	envelope := messageEnvelope{
+		DestinationTopic: destinationTopic,
+		UUID:             msg.UUID,
+		Payload:          msg.Payload,
+		Metadata:         msg.Metadata,
+	}
+
+	envelopedMessage, err := json.Marshal(envelope)
```

Comment:
it probably would be nice to abstract serialization method (with json default :) )

Reply:
but maybe it's not really needed :P probably nobody is interested about transport here